pkg/syncer/util.go

```diff
@@ -884,15 +888,26 @@ func PatchStoragePolicyUsage(ctx context.Context, cnsOperatorClient client.Clien
 		return err
 	}
 	rawPatch := client.RawPatch(apitypes.MergePatchType, patch)
-	err = cnsOperatorClient.Patch(ctx, oldObj, rawPatch)
-	log.Debugf("Patching the StoragePolicyUsageCR %q on namespace: %q with the quota usage data: %+v",
-		oldObj.Name, oldObj.Namespace, newObj.Status.ResourceTypeLevelQuotaUsage)
-	if err != nil {
-		log.Errorf("failed to patch StoragePolicyUsage instance: %q on namespace: %q. Error: %+v",
-			oldObj.Name, oldObj.Namespace, err)
-		return err
+
+	// Try to patch StoragePolicyUsage CR for allowedRetries times
+	allowedRetries := allowedRetriesToPatchStoragePolicyUsage
+	attempt := 0
+	for {
+		attempt++
+		err = cnsOperatorClient.Patch(ctx, oldObj, rawPatch)
+		if err != nil && attempt >= allowedRetries {
```

Comment:
@deepakkinni If this still fails after retries, will it be reconciled by full sync?

Reply:
Not yet. There is a pending task for CsiFullSync to fix StoragePolicyUsage "usage" based on the sum of CnsVolumeInfo aggregated snapshot size. I just added it.

Reply:
Ok.